plugin/trino-elasticsearch/src/main/java/io/trino/plugin/elasticsearch/ElasticsearchMetadata.java

```diff
@@ -543,17 +544,22 @@ public ConnectorTableProperties getTableProperties(ConnectorSession session, Con
         for (ConnectorExpression expression : expressions) {
             if (expression instanceof Call) {
                 Call call = (Call) expression;
-                // TODO Support ESCAPE character when it's pushed down by the engine
-                if (LIKE_PATTERN_FUNCTION_NAME.equals(call.getFunctionName()) && call.getArguments().size() == 2 &&
-                        call.getArguments().get(0) instanceof Variable && call.getArguments().get(1) instanceof Constant) {
+
+                if (isLikeCall(call)) {
+                    Optional<Slice> escape = Optional.empty();
```

Comment:
move this after `Object pattern =` line, so that you process arguments in order: value, pattern, scape.